doc/guidelines/development/docker/index.rst

```diff
@@ -2,10 +2,113 @@
+============================
+
+First, you need to download "docker image". For example, lsstts develop-env can be installed following the instructions from here: `lsstts docker image <https://hub.docker.com/r/lsstts/develop-env/tags>`_ 
+
+.. image:: ../../../images/install_docker_image.png
+    :width: 700px
+
+You can copy the docker pull command and run it from the terminal.
+
+Now you will see image installed in your Docker app like the image below. 
+
+.. image:: ../../../images/docker_images.png
+  :width: 700px
+
+There are two ways to open the container from the image you just installed. You can click the play button to start the container, or put command in terminal to run.
+
+.. image:: ../../../images/open_container_playbutton.png
+  :width: 700px
+
+When opening the container using the play button, there are options you can add. You can put container name, host, volumes, and environmental variables. If you don't give a container name, it will randomly create name for you. 
+
+Similarly, when you run the container using the command, you can select options. For example:
+
+::
+
+    docker run --name lsst-dev -p 80:80 --volume ${place_where_I_save_repos}:/home/saluser/develop -it lsstts/develop-env:develop
+
+Here "--name" is the name of the container, "-p" means port, "--volume" as volumne, where you saved your repo, and "-it" as the image you are using.
+Name your repo path as "/home/saluser/develop" to reduce confusion.
+Basic idea of giving "--volume" and "-it" is that you are creating container based on the "image", and also making it to connect to your repo, which can be the git cloned repository.
+This way you can make the container and repo to communicate, testing runs in the container, update your repo by copying or movnig the files from the container, and then you can push your local repository to the origin.
+
+The containers created will show up when you click "containers" on the left side of the docker desktop as below. You can stop, or delete the containers with the icons. 
+
+.. image:: ../../../images/containers_indocker.png
+  :width: 700px
+
+Opening Docker in Visual Studio Code
+=====================================
+Visual Studio Code makes it easier to use docker environment literally more visual, and straight forward to understand. It helps you to find the files like a finder, to let you able to open the files from double clicking. 
+
+When you open the Visual Studio Code and go to the docker icon (whale shape) on the left column, you will see the list of images installed and containers you opened.  
+
+.. figure:: ../../../images/visual_studio_docker.png
+    :width: 400px
+
+Right click while pointing on the container you want to use, and select "attach Visual Studio Code". It will open your container into the new window.
+
+In the local terminal where you just ran the command to open the container, or from the terminal in the visual studio code, run ::
+    
+    bash ~/.setup_dev.sh
```

Comment:
I think the best way to run this is by using the source command instead of bash, as it will assure that the script will be run even if has been written in csh (or any other shell language):

`source ~/.setup_dev.sh`